src/Annotator.js

```diff
@@ -274,16 +283,24 @@ class Annotator extends EventEmitter {
                 <button class="${CLASS_DIALOG_CLOSE}">${ICON_CLOSE}</button>
             </div>`.trim();
 
-        this.container.appendChild(this.mobileDialogEl);
+        // Only append the mobile dialog once
+        if (!this.container.querySelector(`.${CLASS_MOBILE_ANNOTATION_DIALOG}`)) {
+            this.container.appendChild(this.mobileDialogEl);
```

Comment:
Could we add the check on whatever calls setupMobileDialog? It's small but you you're still doing all of that DOM manipulation above multiple times even if appending it gets blocked.